primer3/thermoanalysis.pyx

```diff
@@ -47,51 +47,39 @@ from libc.stdlib cimport (
 from libc.string cimport strlen
 
 import atexit
-import os.path
+import os.path as op
+import sys
+import threading
+import warnings as pywarnings
 from typing import (
     Any,
     Dict,
+    Optional,
     Union,
 )
 
-from .argdefaults import Primer3PyArguments
+from primer3 import argdefaults
 
-DEFAULT_P3_ARGS = Primer3PyArguments()
 _DID_LOAD_THERM_PARAMS = False
+_DEFAULT_WORD_LEN_2 = 16  # see masker.h
+DEFAULT_P3_ARGS = argdefaults.Primer3PyArguments()
+SNAKE_CASE_DEPRECATED_MSG = 'Function deprecated please use "%s" instead'
 
-# ~~~~~~~~~~~~~~~~~~~~~~~~~ External C declarations ~~~~~~~~~~~~~~~~~~~~~~~~~ #
+CALL_THREAD_LOCK = threading.Lock()
```

Comment:
I'd throw a comment on here explaining the necessity for this lock and the fact that it should be obsoleted by planned thread safety-related changes in `v1.1.x`

Reply:
added